bin/recovery.c

```diff
@@ -909,28 +955,46 @@ PageInit(Page page, Size pageSize, Size specialSize)
  * treat such a page as empty and without free space.  Eventually, VACUUM
  * will clean up such a page and make it usable.
  */
+
+#if PG_VERSION_NUM >= 160000
 bool
-PageHeaderIsValid(PageHeader page)
+PageHeaderIsValid(Page page)
+#else
+bool
+PageHeaderIsValid(PageHeader phdr)
+#endif
 {
 	char	   *pagebytes;
 	int			i;
+#if PG_VERSION_NUM >= 160000
+	PageHeader phdr = (PageHeader) page;
+#endif
 
 	/*
 	 * Check normal case
 	 */
-	if (PageGetPageSize(page) == BLCKSZ &&
```

Comment:
We can't remove the lines for under v16.

Reply:
No, I did not delete this line. I just changed the variable name from page to phdr. 
And make sure PageGetPageSize and PageGetPageLayoutVersion can use Page structure to get values.

Reply:
Is it your expected to remove the following lines?

```
		page->pd_lower >= SizeOfPageHeaderData &&
		page->pd_lower <= page->pd_upper &&
		page->pd_upper <= page->pd_special &&
		page->pd_special <= BLCKSZ &&
		page->pd_special == MAXALIGN(page->pd_special))
```

Reply:
No, it is not a remove. I changed the code.